src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBuffer.cs

```diff
@@ -1027,6 +1027,16 @@ internal void SetToString(string value)
             _isNull = false;
         }
 
+#if netcoreapp
```

Comment:
It's recommended to separate these netcoreapp specific methods to their own files such as SqlBuffer.netcoreapp.cs 
You can have a look at this pr for reference:
https://github.com/dotnet/corefx/pull/29697

Reply:
The method is netcore specific because that is the only configuration that includes span. If other configurations started to include span (which I would hope they will for performance reasons) then this version could be used. Really what would best is a feature specific #define.

The methods are also direct replacements not additions so putting them side by side allows editors visibility of both versions if anything needs to change. 

I can make SqlBuffer partial and move the span using versions into a netcore specific file but it won't remove the #defines in the main file, those are still needed to turn off the old way and prevent the unised methods being compiled in causing bloat.

Reply:
Bearing in mind my feedback above, do you still want me to make these changes?

Reply:
@Wraith2 I'd suggest moving the new methods to netcoreapp specific files and leave the #defines inside main file be as is.  This is something we discussed earlier for the PR I referenced, and the decision was to try and reduce` #if netcoreapp`   as much as possible.

Reply:
Done, please review and feedback any further changes.

Reply:
@Wraith2 Thanks! There are some build failures on the CI. For example:
```System\Data\SqlClient\SqlBuffer.netcoreapp.cs(40,13): error CS0103: The name '_isNull' does not exist in the current context```
Try building using this command to build all configurations and see errors locally: ```build -allConfigurations```